src/main/java/org/eclipse/cargotracker/domain/model/cargo/Cargo.java

```diff
@@ -52,9 +52,13 @@
  */
 @Entity
 @NamedQuery(name = "Cargo.findAll", query = "Select c from Cargo c")
+@NamedQuery(name = "Cargo.findAllWithItineraryLegs", query = "Select c from Cargo c join fetch c.itinerary.legs")
```

Comment:
@hantsy, I have tested this carefully and in EclipseLink this is causing duplicate cargo instances to be returned. Can you kindly investigate a bit more what JPA solution would work with DDD, EclipseLink and Hibernate? I would also like to understand what is preventing you from testing on Payara properly. Is the issue simply phone-home that you can easily turn off: https://docs.payara.fish/community/docs/5.194/documentation/payara-server/phone-home/disabling-phone-home.html? Is there some other issue you are having with Payara? Have you looked for a solution on the Payara Forums: https://groups.google.com/g/payara-forum? Also, have you tried the EclipseLink and Hibernate forums?

Reply:
1. Just tried it on the Wildfly server, maybe add a `distinct in the clause?
2. Currently I am on some mail list(including WildFly) and also in WildFly zulipchat. When JBoss forum became readonly, I moved to Stackoverflow when I have some questions about JBosss products. Thanks for introducing the payara forum to me, just join it. In fact, in the past years, I rarely used Glassfish/Payara in real-world projects, most of the time, it is used as target testing platform. JBoss/WildFly was used in several former proejcts.

Reply:
Kindly investigate this a bit more using the appropriate forums. I don't think adding a DISTINCT clause is the right answer for our code base. There has to be a way that is portable and does not feel like too much of a hack.